maths/segmented_sieve.py

```diff
@@ -4,7 +4,22 @@
 
 
 def sieve(n: int) -> list[int]:
-    """Segmented Sieve."""
+    """
+    Segmented Sieve.
+
+    Examples:
+    >>> sieve(2**3)
+    [2, 3, 5, 7]
+
+    >>> sieve(3**3)
+    [2, 3, 5, 7, 11, 13, 17, 19, 23]
+
+    >>> sieve()
+    Traceback (most recent call last):
+        ...
+    TypeError: sieve() missing 1 required positional argument: 'n'
+

```

Comment:
```suggestion
```
I think it's fine to assume that the function will be called on the right number of inputs, so I don't think this test is necessary